### PR TITLE
Disable payment button if term checkbox is unchecked

### LINKED ISSUE #19161: Confirmation button on Checkout should be disabled if Terms and conditions are not approved
#### Describe the bug

On checkout, with PaymentOption of type Binary, I don't approve Terms and conditions but I can validate my order.

#### Expected behavior

On checkout, if Terms and conditions is not approved, all confirmation buttons should be disabled.

#### Steps to Reproduce

1. Install a payment module with a PaymentOption of type Binary.
For example : [module Payment Example v2.0.0](https://github.com/PrestaShop/paymentexample/pull/14) or [module Atos](https://addons.prestashop.com/en/payment-card-wallet/1-sips-10-worldline-atos-15-16-17.html)
2. Add a Product to Cart
1. Go to Cart
2. Click on Proceed to checkout
3. On step Payment, don't approve Terms and conditions
4. See confirmation button seems disabled
5. Click on confirmation button
6. See Order confirmation page

**Screenshots**

checkout-payment-confirm

#### Additional information

* PrestaShop version: 1.7.6
* PHP version: 7.2

**Epic** : #17034

### PR DESCRIPTION
**Pull request by NeOMakinG**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Binary payment button was enabled while terms checkbox was unchecked, it's a problem
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes #19161.
| How to test?  | Install the module example provided by @Matt75 in #19161, and then go to checkout page, select binary, the "Pay binary" button shouldn't be clickable, also test this with Atos module as you need to verify that inputs are disabled too instead of buttons

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19438)

### CODE REVIEW

File: themes/_core/js/checkout-payment.js [hunk 1]
```diff
@@ -85,6 +85,7 @@ class Payment {
       var paymentOption = this.getPaymentOptionSelector(selectedOption);
       this.hideConfirmation();
       $(paymentOption).show();
+      $(paymentOption + ' button').attr('disabled', !show);
```
Comment (Progi1984) [inline themes/_core/js/checkout-payment.js]:
```suggestion
      $(`${paymentOption} button`).attr('disabled', !show);
```

**Review by Progi1984 — COMMENTED** [review]
A small change

File: themes/_core/js/checkout-payment.js (hunk 1 above)
Comment (Matt75) [inline themes/_core/js/checkout-payment.js]: https://github.com/PrestaShop/PrestaShop/issues/19161#issuecomment-629316433

**Review by Matt75 — CHANGES_REQUESTED** [review]

**Review by Matt75 — DISMISSED** [review]

**Review by Progi1984 — DISMISSED** [review]

**Review by PierreRambaud — DISMISSED** [review]

**Comment by Robin-Fischer-PS** [comment]
Hi @NeOMakinG !

Now, even when terms are checked, the "Pay" button is still disabled, so I cannot pay with Binary : 

screencapture-localhost-psdevelop-develop-fr-commande-2020-05-29-15_57_53

File: themes/_core/js/checkout-payment.js [hunk 2]
```diff
@@ -85,6 +85,14 @@ class Payment {
       var paymentOption = this.getPaymentOptionSelector(selectedOption);
       this.hideConfirmation();
       $(paymentOption).show();
+      console.log(show);
```
Comment (Matt75) [inline themes/_core/js/checkout-payment.js]:
```suggestion

```

**Review by Matt75 — CHANGES_REQUESTED** [review]

File: themes/_core/js/checkout-payment.js (hunk 2 above)
Comment (NeOMakinG) [reply themes/_core/js/checkout-payment.js]: wow

Comment (NeOMakinG) [reply themes/_core/js/checkout-payment.js]: You didn't see anything 🤫

**Review by Matt75 — APPROVED** [review]

**Review by PierreRambaud — APPROVED** [review]

**Review by Progi1984 — APPROVED** [review]

**Comment by Progi1984** [comment]
Thanks @NeOMakinG

**Pull request merged**